src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -1,12 +1,42 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import
 
+import json
+import cherrypy
+
+from mgr_module import CommandResult
 from . import ApiController, AuthRequired, RESTController
 from .. import mgr
+from ..services.ceph_service import CephService
 
 
 @ApiController('host')
 @AuthRequired()
 class Host(RESTController):
     def list(self):
         return mgr.list_servers()
+
+
+@ApiController('host/scrub')
+@AuthRequired()
+class HostScrub(RESTController):
+    @cherrypy.expose
+    @cherrypy.tools.json_out()
+    def create(self, hosts, deep_scrub):
+        api_scrub = "deep_scrub {}" if deep_scrub else "scrub {}"
+        services = CephService.get_service_map("osd")
+        osds = []
+
+        result = CommandResult("")
+        for host in hosts:
+            for osd in services[host]['services']:
+                argdict = {
+                    "prefix": api_scrub.format(osd),
+                    "format": "json",
+                }
+
+                mgr.send_command(result, "mon", "", json.dumps(argdict), "")
+                # CephService.send_command("osd", api_scrub.format(osd['id']))
```

Comment:
Why did you chose not to use the CephService.send_command method?
Using that method would take care of raising errors appropriately and it would check the result of the operation. Both are currently unhandled in your implementation.

Reply:
When I tried it with CephService the application just hang and nothing happened.
But I will try it again, since it looks like I was using the wrong 1st parameter...

Reply:
Done.